GetIntoTeachingApiTests/Models/Crm/BaseModelTests.cs

```diff
@@ -332,6 +332,17 @@ public void ToEntity_WithNew()
             mockEntity.GetAttributeValue<string>("dfe_field3").Should().Be(mock.Field3);
         }
 
+        [Fact]
+        public void ToEntity_WhenIdIsApiGenerated_CallsNewEntity()
+        {
+            var mock = new ICanGenerateUpfrontIdMock();
```

Comment:
If we have `BaseModel` implement the interface I think we could just use that model directly, or one of the subclasses to avoid having to mock it?